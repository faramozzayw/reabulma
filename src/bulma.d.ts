export declare namespace Bulma {
	export type Colors =
		| "white"
		| "light"
		| "dark"
		| "black"
		| "primary"
		| "info"
		| "success"
		| "warning"
		| "danger"
		| "text";

	export interface Color {
		isColor?: Colors;
	}

	export type Sizes = "small" | "normal" | "medium" | "large";
	export interface Size {
		isSize?: Sizes;
	}

	export interface Tag {
		tag?: keyof React.ReactHTML;
	}

	export interface Loading {
		isLoading?: boolean;
	}

	export interface Light {
		isLight?: boolean;
	}

	export interface Rounded {
		isRounded?: boolean;
	}

	export type Align = "left" | "centered" | "right";

	export interface Alignment {
		isAlign?: Align;
	}

	export interface Active {
		isActive?: boolean;
	}

	export type HeadingSizes =
		| 1
		| "1"
		| 2
		| "2"
		| 3
		| "3"
		| 4
		| "4"
		| 5
		| "5"
		| 6
		| "6";

	export interface Heading {
		isSize?: HeadingSizes;
		isSpaced?: boolean;
	}

	export type States = "hovered" | "focused" | "active";
	export interface State {
		isState?: States;
	}

	export interface FullWidth {
		isFullWidth?: boolean;
	}
}
