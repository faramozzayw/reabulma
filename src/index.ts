export type { Bulma } from "./bulma";

// layout
export { Hero, HeroBody, HeroHeader, HeroFooter } from "./layout/Hero/index";
export { Tile } from "./layout/Tile";
export { Container } from "./layout/container";

// elements
export { Button, Buttons } from "./elements/Button/index";
export { Table, TableContainer } from "./elements/Table/index";
export { Box } from "./elements/box";
export { Title } from "./elements/title";
export { Content } from "./elements/content";
export { Notification } from "./elements/notification";
export { ProgressBar } from "./elements/ProgressBar";
export { Block } from "./elements/block";
export { Tag } from "./elements/tag";
export { Image } from "./elements/image";
export { Delete } from "./elements/delete";

// forms
export { Control } from "./form/control";
export { Label } from "./form/label";
export { Select } from "./form/select";
export { TextArea } from "./form/textarea";
export { Help } from "./form/help";
// forms / field
export { Field } from "./form/field";
export { FieldLabel } from "./form/FieldLabel";

// components
export {
	Panel,
	PanelHeading,
	PanelBlock,
	PanelIcon,
} from "./components/Panel/index";
export {
	Navbar,
	NavbarMenu,
	NavbarBurger,
	NavbarBrand,
	NavbarItem,
	NavbarLink,
	NavbarDivider,
	NavbarDropdown,
	NavbarStart,
	NavbarEnd,
} from "./components/Navbar/index";
export {
	Card,
	CardHeader,
	CardHeaderTitle,
	CardImage,
	CardContent,
	CardFooter,
	CardFooterItem,
} from "./components/Card/index";

export {
	Modal,
	ModalBackground,
	ModalContent,
	ModalClose,
} from "./components/Modal/index";

export { Tabs } from "./components/Tabs/Tabs";
export { Tab } from "./components/Tabs/Tab";
