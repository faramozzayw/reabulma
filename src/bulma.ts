import { fractions, width } from "./columns/utils";

export declare namespace Bulma {
	type Theme = "light" | "dark";
	type Viewports = "mobile" | "tablet" | "desktop" | "widescreen" | "fullhd";
	type IsOnly = "-only" | "";

	export type Colors =
		| "white"
		| "light"
		| "dark"
		| "black"
		| "primary"
		| "info"
		| "success"
		| "warning"
		| "danger"
		| "text";

	export type GrayShades =
		| "black-bis"
		| "black-ter"
		| "grey-darker"
		| "grey-dark"
		| "grey"
		| "grey-light"
		| "grey-lighter"
		| "white-ter"
		| "white-bis";

	export interface Color {
		isColor?: Colors;
	}

	export type Sizes = "small" | "normal" | "medium" | "large";
	export interface Size {
		isSize?: Sizes;
	}

	export interface Tag {
		tag?: keyof React.ReactHTML;
	}

	export interface Loading {
		isLoading?: boolean;
	}

	export interface Light {
		isLight?: boolean;
	}

	export interface Rounded {
		isRounded?: boolean;
	}

	export interface Static {
		isStatic?: boolean;
	}

	export interface Link {
		isLink?: boolean;
	}

	export type Align = "left" | "centered" | "right";

	export interface Alignment {
		isAlign?: Align;
	}

	export interface Active {
		isActive?: boolean;
	}

	export type ImageRatioList =
		| "square"
		| "1by1"
		| "5by4"
		| "4by3"
		| "3by2"
		| "5by3"
		| "16by9"
		| "2by1"
		| "3by1"
		| "4by5"
		| "3by4"
		| "2by3"
		| "3by5"
		| "9by16"
		| "1by2"
		| "1by3";

	export interface ImageRatio {
		isRatio?: ImageRatioList;
	}

	export type ImageSizeList =
		| "128x128"
		| "16x16"
		| "24x24"
		| "32x32"
		| "48x48"
		| "64x64"
		| "96x96";

	export interface ImageSize {
		isSize?: ImageSizeList;
	}

	export type HeadingSizes =
		| 1
		| "1"
		| 2
		| "2"
		| 3
		| "3"
		| 4
		| "4"
		| 5
		| "5"
		| 6
		| "6";

	export interface Heading {
		isSize?: HeadingSizes;
		isSpaced?: boolean;
	}

	export type States = "hovered" | "focused" | "active";
	export interface State {
		isState?: States;
	}

	export interface FullWidth {
		isFullWidth?: boolean;
	}

	export namespace Columns {
		export type RegularSizes = 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12;
		export type StringSizes =
			| "1"
			| "2"
			| "3"
			| "4"
			| "5"
			| "6"
			| "7"
			| "8"
			| "9"
			| "10"
			| "11"
			| "12";

		export type Fractions = keyof typeof fractions;
		export type Width = keyof typeof width;

		type AllSizes = RegularSizes | StringSizes | Fractions | Width;

		export type SizeObject = {
			[key in Viewports | "default"]?: AllSizes;
		};

		export type Platforms = keyof SizeObject;

		export interface AbstructSize {}

		export interface Size extends AbstructSize {
			isSize?: AllSizes | SizeObject;
		}

		export interface Offset extends AbstructSize {
			isOffset?: RegularSizes | StringSizes | Fractions | SizeObject;
		}
	}

	export namespace Helpers {
		export interface AllHelpers
			extends TextColor,
				BackgroundColor,
				FullWidth,
				Spacing,
				Show,
				Hide,
				VisibilityHelpers {
			flexbox?: Flexbox;
			typography?: Typography;
			/** Moves an element `left` or `right` */
			isPulled?: "left" | "right";
			/** Fixes an element's floating children */
			isClearFix?: boolean;
			/** Completely **covers** the first positioned parent */
			isOverlay?: boolean;
			/** Adds overflow **hidden** */
			isClipped?: boolean;
			/** Removes any **radius** */
			isRadiusless?: boolean;
			/** Removes any **shadow** */
			isShadowless?: boolean;
			/** Prevents the text from being **selectable** */
			isUnselectable?: boolean;
			/** Applies `cursor: pointer !important` to the element. */
			isClickable?: boolean;
			/** Applies `position: relative` to the element. */
			isRelative?: boolean;
		}

		/*
		**************************
			Flexbox
		**************************
		*/
		export interface Flexbox {
			alignContent?: AlignContent;
			alignItems?: AlignItems;
			justifyContent?: JustifyContent;
			grow?: 1 | 2 | 3 | 4 | 5 | 6;
			shrink?: 1 | 2 | 3 | 4 | 5 | 6;
			wrap?: FlexWrap;
			direction?: FlexDirection;
		}

		export type JustifyContent =
			| "flex-start"
			| "flex-end"
			| "center"
			| "space-between"
			| "space-around"
			| "space-evenly"
			| "start"
			| "end"
			| "left"
			| "right";

		export type AlignContent =
			| "flex-start"
			| "flex-end"
			| "center"
			| "space-between"
			| "space-around"
			| "space-evenly"
			| "start"
			| "end"
			| "stretch"
			| "baseline";

		export type AlignItems =
			| "stretch"
			| "flex-start"
			| "flex-end"
			| "center"
			| "baseline"
			| "start"
			| "end"
			| "self-start"
			| "self-end";

		export type FlexWrap = "nowrap" | "wrap" | "wrap-reverse";
		export type FlexDirection =
			| "row"
			| "row-reverse"
			| "column"
			| "column-reverse";

		/*
		**************************
			Typography
		**************************
		*/
		export interface Typography {
			/** You can transform the text with the use of one of 4 text transformation helpers */
			transformation?: TextTransformation;
			/** You can transform the text weight with the use of one of 5 text weight helpers */
			weight?: TextWeight;
			/** You can change the font family with the use of one of 5 font family helpers */
			family?: Family;
			size?: Sizes | ResponsiveSize;
			align?: TextAlign | ResponsiveTextAlignment;
		}

		export type Sizes = 1 | 2 | 3 | 4 | 5 | 6 | 7;
		export type ResponsiveSize = {
			[key in Viewports | "touch" | "default"]?: Sizes;
		};

		export type Family =
			| "sans-serif"
			| "monospace"
			| "primary"
			| "secondary"
			| "code";

		export type TextWeight =
			| "light"
			| "normal"
			| "medium"
			| "semibold"
			| "bold";

		export type TextTransformation =
			| "capitalized"
			| "lowercase"
			| "uppercase"
			| "italic";

		export interface TextAlignment {
			isAlign?: Align | "justified";
		}

		export type TextAlign = Align | "justified";
		// prettier-ignore
		export type ResponsiveTextAlignment = {
			[key in Viewports | "touch" | "default"]?: TextAlign | [TextAlign?, boolean?];
		};

		/*
		**************************
			Colors
		**************************
		*/
		export type AllColors =
			| GrayShades
			| Exclude<Colors, "text">
			| `${Exclude<Colors, "text" | "light" | "dark">}-${Theme}`;

		export interface TextColor {
			hasTextColor?: AllColors;
		}

		export type BackgroundColors =
			| GrayShades
			| Exclude<Colors, "text">
			| `${Exclude<Colors, "text" | "light" | "dark">}-${Theme}`;

		export interface BackgroundColor {
			hasBackgroundColor?: BackgroundColors;
		}

		/*
		**************************
			Spacing
		**************************
		*/
		export type SpacingType = "m" | "p";
		export type SpacingDirection = "t" | "b" | "r" | "l" | "x" | "y" | "";
		export type SpacingSize = 0 | 1 | 2 | 3 | 4 | 5 | 6;
		export type AllSpacing = `${SpacingType}${SpacingDirection}-${SpacingSize}`;

		export interface Spacing {
			/** Bulma provides margin `m*` and padding `p*` helpers in all directions */
			spacing?: AllSpacing[];
		}
		/*
		**************************
			Visibility
		**************************
		*/

		export type DisplayType =
			| "block"
			| "flex"
			| "inline"
			| "inline-block"
			| "inline-flex";

		export type Platforms = `${Viewports | "touch"}${IsOnly}`;
		export type PlatformsWithDefault = Platforms | "default";

		export type DisplayObject = {
			[key in DisplayType]?: PlatformsWithDefault | PlatformsWithDefault[];
		};
		export type Displays = `${DisplayType}-${Platforms}`;
		// export type DisplayObject = {
		// 	[key in DisplayType]?: `${Extract<DisplayType, key>}-${Displays}`;
		// };

		export interface Show {
			isDisplay?: Displays | Displays[] | DisplayObject;
		}

		export interface Hide {
			/** Hides element */
			isHidden?: boolean | Platforms | Platforms[];
		}

		export interface VisibilityHelpers {
			/** Adds visibility **hidden** */
			isInvisible?: boolean;
			/** Hide elements **visually** but keep the element available to be announced by a **screen reader** */
			isSrOnly?: boolean;
		}
	}
}
