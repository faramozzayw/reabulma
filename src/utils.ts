import { Bulma } from "./bulma";

export function getModifiersCreator<T>(key: string) {
	return (props: { [key: string]: T | undefined }) => {
		const modifier = props[key];

		return modifier ? { [`is-${modifier}`]: true } : {};
	};
}

export const getColorModifiers = getModifiersCreator<Bulma.Colors>("isColor");
export const getAlignmentModifiers = getModifiersCreator<Bulma.Align>(
	"isAlign",
);
export const getSizeModifiers = getModifiersCreator<Bulma.Sizes>("isSize");
export const getStateModifiers = getModifiersCreator<Bulma.States>("isState");

export const getActiveModifiers = ({ isActive }: Bulma.Active) => {
	return isActive ? { [`is-active`]: true } : {};
};

export const getLoadingModifiers = ({ isLoading }: Bulma.Loading) => {
	return isLoading ? { [`is-loading`]: true } : {};
};

export const getHeadingModifiers = ({
	isSize: size = 1,
	isSpaced,
}: Bulma.Heading) => {
	const isSize = { [`is-${size}`]: true };

	return {
		...isSize,
		"is-spaced": isSpaced,
	};
};
