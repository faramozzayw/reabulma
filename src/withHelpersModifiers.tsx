import React from "react";
import classNames from "classnames";

import { Bulma } from "./bulma";
import {
	getBackgroundColorModifiers,
	getFlexboxModifiers,
	getSpacingModifiers,
	getTextAlignmentModifiers,
	getTextColorModifiers,
} from "./utils";

export type HelpersComponent<T> = React.FC<
	T & Bulma.Helpers.AllHelpers & React.HTMLProps<HTMLElement>
>;
export type Component<T> = T extends React.FC ? T : never;

export function getDisplayName<T>(WrappedComponent: React.ComponentType<T>) {
	return WrappedComponent.displayName ?? WrappedComponent.name ?? "Component";
}

export function withHelpersModifiers<T>(Component: React.ComponentType<T>) {
	const WrappedComponent: HelpersComponent<T> = ({
		flexbox,
		hasTextColor,
		hasBackgroundColor,
		spacing,
		isPulled,
		isClearFix,
		isOverlay,
		isClipped,
		isRadiusless,
		isShadowless,
		isUnselectable,
		isClickable,
		isRelative,
		isAlign,
		...props
	}) => {
		const className = classNames(
			{
				[`is-pulled-${isPulled}`]: Boolean(isPulled),
				"is-clearfix": isClearFix,
				"is-overlay": isOverlay,
				"is-clipped": isClipped,
				"is-radiusless": isRadiusless,
				"is-shadowless": isShadowless,
				"is-unselectable": isUnselectable,
				"is-clickable": isClickable,
				"is-relative": isRelative,
				//...getSpacingModifiers({ spacing }),
				...getTextAlignmentModifiers({ isAlign }),
				...getFlexboxModifiers(flexbox),
				...getTextColorModifiers({ hasTextColor }),
				...getBackgroundColorModifiers({ hasBackgroundColor }),
			},
			props.className,
		);

		return <Component {...(props as T)} className={className} />;
	};

	WrappedComponent.displayName = `WithHelpers(${getDisplayName(Component)})`;

	return WrappedComponent;
}

/*
export function withHelpersModifiers<P extends object>(
	Component: React.ComponentType<P>,
) {
	return class extends React.Component<P & BulmaHelpers & React.HTMLProps<HTMLElement>> {
		static displayName = `WithHelpers(${getDisplayName(Component)})`;

		render() {
			const { flexbox, ...props } = this.props;

			const className = classNames(
				{
					...getFlexboxModifiers(flexbox),
				},
				props.className,
			);

			return <Component {...props as P} className={className} />;
		}
	};
}
*/
