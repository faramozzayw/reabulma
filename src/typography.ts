import { Bulma } from "./bulma";
import { getModifiersCreator, Modifier, ModifierFunction } from "./utils";

export const getTransformationModifiers = getModifiersCreator<
	Bulma.Helpers.Typography,
	"transformation"
>("transformation");

export const getWeightModifiers = getModifiersCreator<
	Bulma.Helpers.Typography,
	"weight"
>("weight", {
	prefix: "has-text-weight",
});

export const getFamilyModifiers = getModifiersCreator<
	Bulma.Helpers.Typography,
	"family"
>("family", {
	prefix: "is-family",
});

export const getResponsiveSizeModifiers = (
	isSize: Bulma.Helpers.ResponsiveSize,
): Modifier => {
	let resultSize = {};

	for (const [platform, size] of Object.entries(isSize)) {
		const platformPart = platform === "default" ? "" : `-${platform}`;

		resultSize = {
			...resultSize,
			[`is-size-${size}${platformPart}`]: true,
		};
	}

	return resultSize;
};

export const getSizeModifiers: ModifierFunction = (
	size: Bulma.Helpers.Typography["size"],
) => {
	if (typeof size === "number") {
		return {
			[`is-size-${size}`]: true,
		};
	}

	if (typeof size === "object") return getResponsiveSizeModifiers(size);

	return {};
};

export const getPlainAlignModifiers = getModifiersCreator<
	Bulma.Helpers.Typography,
	"align"
>("align", {
	prefix: "has-text",
});

export const getResponsiveAlignModifiers = (
	isAlign: Bulma.Helpers.ResponsiveTextAlignment,
): Modifier => {
	let resultSize = {};

	for (const [platform, align] of Object.entries(isAlign)) {
		const isDefault = platform === "default";
		const isTuple = Array.isArray(align);
		const withOnly = isTuple ? Boolean(align[1]) : false;

		const alignPart = isTuple ? align[0] : align;
		const platformPart = isDefault ? "" : `-${platform}`;
		const onlyPart = withOnly ? "-only" : "";

		resultSize = {
			...resultSize,
			[`has-text-${alignPart}${platformPart}${onlyPart}`]: true,
		};
	}

	return resultSize;
};

export const getTextAlignModifiers: ModifierFunction = (
	align: Bulma.Helpers.Typography["align"],
) => {
	if (typeof align === "string") return getPlainAlignModifiers({ align });
	if (typeof align === "object") return getResponsiveAlignModifiers(align);

	return {};
};

export const getTypographyModifiers = (
	typography: Bulma.Helpers.Typography,
) => ({
	...getTextAlignModifiers(typography?.align),
	...getSizeModifiers(typography?.size),
	...getTransformationModifiers({ transformation: typography?.transformation }),
	...getWeightModifiers({ weight: typography?.weight }),
	...getFamilyModifiers({ family: typography?.family }),
});
