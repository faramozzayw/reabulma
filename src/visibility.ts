import { Bulma } from "./bulma";
import { getModifiersCreator, Modifier, ModifierFunction } from "./utils";

const getStringDisplayModifiers = getModifiersCreator<
	Bulma.Helpers.Show,
	"isDisplay"
>("isDisplay");

const getArrayDisplayModifiers = (
	isDisplay: Bulma.Helpers.Displays[],
	type?: Bulma.Helpers.DisplayType,
) => {
	const typePart = Boolean(type) ? `-${type}` : "";
	let result = new Map<string, boolean>();

	for (const item of isDisplay) {
		result.set(`is${typePart}-${item}`, true);
	}

	return Object.fromEntries(result);
};

const getObjectDisplayModifiers: ModifierFunction = (
	isDisplay: Bulma.Helpers.DisplayObject,
) => {
	const display = Object.entries(isDisplay)
		.map(([type, platfrom]) => {
			if (Array.isArray(platfrom))
				return getArrayDisplayModifiers(
					platfrom as any,
					type as Bulma.Helpers.DisplayType,
				);

			return { [`is-${type}-${platfrom}`]: true };
		})
		.reduce((acc, item) => ({ ...acc, ...item }));

	return display;
};

export const getDisplayModifiers: ModifierFunction = ({
	isDisplay,
}: Bulma.Helpers.Show) => {
	if (typeof isDisplay === "string") {
		return getStringDisplayModifiers({ isDisplay });
	}

	if (Array.isArray(isDisplay)) return getArrayDisplayModifiers(isDisplay);

	if (typeof isDisplay === "object")
		return getObjectDisplayModifiers(isDisplay);

	return {};
};

const getStringHideModifiers = getModifiersCreator<
	Bulma.Helpers.Hide,
	"isHidden"
>("isHidden", {
	prefix: "is-hidden",
});

export const getHideModifiers: ModifierFunction = ({
	isHidden,
}: Bulma.Helpers.Hide) => {
	if (typeof isHidden === "boolean") {
		return { [`is-hidden`]: isHidden };
	}

	if (typeof isHidden === "string") {
		return getStringHideModifiers({ isHidden });
	}

	if (Array.isArray(isHidden)) {
		return isHidden
			.map((item) => getStringHideModifiers({ isHidden: item }))
			.reduce((acc, item) => ({ ...acc, ...item }));
	}

	return {};
};
