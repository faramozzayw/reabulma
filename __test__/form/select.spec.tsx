import React, { useEffect, useState } from "react";
import "@testing-library/jest-dom";
import { fireEvent, render } from "@testing-library/react";
import { Select } from "../../src";

const Form = () => {
	const options = ["dark", "green", "blue", "red"];
	const [color, setColor] = useState(options[0]);

	useEffect(() => console.log("e.currentTarget.value"));

	const handleChange = (e: React.FormEvent<HTMLSelectElement>) => {
		setColor(e.currentTarget.value);
	};

	const optionsList = options.map((value) => (
		<option label={value} key={value}>
			{value}
		</option>
	));

	return (
		<>
			<h1>Your favorite color is {color}</h1>
			<Select onChange={handleChange} value={color} data-testid="select">
				{optionsList}
			</Select>
		</>
	);
};

const setupForm = () => {
	const utils = render(<Form />);
	const select = utils.getByTestId("select");
	const header = utils.getByText(/favorite color/);

	return { header, select, ...utils };
};

describe("Select", () => {
	it("renders correctly", () => {
		const { baseElement, container } = render(<Select />);

		expect(container.firstChild).toHaveClass("select");
		expect(baseElement).toMatchSnapshot();
	});

	it("`isLoading` prop", () => {
		const { container: selectWithLoading } = render(<Select isLoading />);
		const { container: selectWithoutLoading } = render(
			<Select isLoading={false} />,
		);

		expect(selectWithLoading.firstChild).toHaveClass("is-loading");
		expect(selectWithoutLoading.firstChild).not.toHaveClass("is-loading");
	});

	it("isState", () => {
		const { container } = render(
			<Select isState="focused" data-testid="focused" />,
		);

		expect(container.firstChild).toHaveClass("is-focused");
	});

	it("isColor", () => {
		const { container } = render(
			<Select isColor="primary" placeholder="primary" />,
		);

		expect(container.firstChild).toHaveClass("is-primary");
	});

	it("`isSize` prop", () => {
		const { container } = render(<Select isSize="medium" />);
		expect(container.firstChild).toHaveClass("is-medium");
	});

	it("input work correctly", () => {
		const { header, getByText, getByTestId } = setupForm();

		expect(header.textContent).toContain("dark");

		fireEvent.change(getByTestId("select"), { target: { value: "red" } });
		expect(header.textContent).toContain("red");

		fireEvent.change(getByTestId("select"), { target: { value: "blue" } });
		expect(header.textContent).toContain("blue");
	});
});
