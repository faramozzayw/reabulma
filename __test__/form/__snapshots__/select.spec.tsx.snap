// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select renders correctly 1`] = `
<body>
  <div>
    <div
      class="select"
    >
      <select
        class=""
      />
    </div>
  </div>
</body>
`;
